kvbc/tools/db_editor/include/kv_blockchain_db_editor.hpp

```diff
@@ -898,6 +927,180 @@ struct ResetMetadata {
+    std::unordered_set<std::string> columnFamilies;
+    if (args.values.size() > 0) {
+      for (const auto &s : args.values) {
+        columnFamilies.insert(s);
+      }
+    } else {
+      columnFamilies = adapter.db()->columnFamilies();
+    }
+
+    std::ostringstream oss;
+
+    oss << "{";
+
+    for (auto columnFamily : columnFamilies) {
+      oss << "\n\t\"" << columnFamily << "\" : {";
+      auto handle = adapter.db()->columnFamilyHandle(columnFamily);
+      uint64_t v;
+      for (auto property : properties) {
+        adapter.db()->rawDB().GetIntProperty(handle, rocksdb::Slice(property->c_str(), property->length()), &v);
+        oss << "\n\t\t\"" << *property << "\" : " << v << ",";
+      }
+
+      oss << "\n\t},";
+    }
+    oss << "\n}";
+    return oss.str();
+  }
+};
+
+struct GetColumnFamiliesStatsDetail {
+  const bool read_only = true;
+  std::string description() const {
+    return "GetTablesStatsDetail\n"
+           " Returns json with a map column family name to these values KV Count, Data Size, Key Size, Value Size\n";
+  }
+
+  std::string toJson(
+      const std::vector<std::pair<std::string, std::unique_ptr<rocksdb::TableProperties>>> &output) const {
+    std::ostringstream oss;
+    oss << "{";
+    for (const auto &itPair : output) {
+      oss << "\n\t\"" << itPair.first << "\" : {";
+      /* KV Count, Data Size, Key Size, Value Size for the column family */
+      {
+        oss << "\n\t\t\"KV count\" : " << itPair.second->num_entries;
+        oss << ",\n\t\t\"Data Size\" : " << itPair.second->data_size;
+        oss << ",\n\t\t\"Key Size\" : " << itPair.second->raw_key_size;
+        oss << ",\n\t\t\"Value Size\" : " << itPair.second->raw_value_size;
+      }
+      oss << "\n},";
+    }
+    oss << "\n}\n";
+    return oss.str();
+  }
+
+  void ResetTableProperties(rocksdb::TableProperties *tp) const {
+    tp->data_size = 0;
+    tp->index_size = 0;
+    tp->filter_size = 0;
+    tp->raw_key_size = 0;
+    tp->raw_value_size = 0;
+    tp->num_data_blocks = 0;
+    tp->num_entries = 0;
+    tp->num_deletions = 0;
+    tp->num_merge_operands = 0;
+    tp->num_range_deletions = 0;
+  }
+
+  void ParseTablePropertiesString(std::string &tp_string, rocksdb::TableProperties *tp) const {
+    double dummy_double;
+    std::replace(tp_string.begin(), tp_string.end(), ';', ' ');
+    std::replace(tp_string.begin(), tp_string.end(), '=', ' ');
+    ResetTableProperties(tp);
+    sscanf(tp_string.c_str(),
+           "# data blocks %" SCNu64 " # entries %" SCNu64 " # deletions %" SCNu64 " # merge operands %" SCNu64
+           " # range deletions %" SCNu64 " raw key size %" SCNu64
+           " raw average key size %lf "
+           " raw value size %" SCNu64
+           " raw average value size %lf "
+           " data block size %" SCNu64 " index block size (user-key? %" SCNu64 ", delta-value? %" SCNu64 ") %" SCNu64
+           " filter block size %" SCNu64,
```

Comment:
I am a bit reluctant about parsing it as that format is, I guess, subject to change. I would say we can maybe leave the `GetColumnFamilyStats` for now only.

Reply:
done